docs/reference/modules/scripting/security.asciidoc

```diff
@@ -1,67 +1,108 @@
+overwhelm the Elasticsearch cluster and bring it down. For example:
+
+Good:
+* Users type text into a search box and the text is sent directly to a
+<<query-dsl-match-query>>, <<query-dsl-match-query-phrase>>,
+<<query-dsl-simple-query-string-query>>, or any of the <<search-suggesters>>.
+* Running a script with any of the above queries that was written as part of
+the application development process.
+* Running a script with `params` provided by users.
+* User actions makes documents with a fixed structure.
+
+Bad:
+* Users can write arbitrary scripts, queries, `_search` requests.
+* User actions make documents with structure defined by users.
 
-[source,yaml]
-----------------
-script.engine.painless.inline.update: true
-----------------
+[float]
+=== Do not weaken script security settings
+By default Elasticsearch allows inline scripting for sandboxed languages,
+namely the scripting language Painless, the template language Mustache, and
+the expression language Expressions. These *ought* to be safe to expose to
+trusted users and to your application servers because they have strong security
+sandboxes. Non-sandboxed languages are disabled by default and enabling them
+is a poor choice because:
+1. This drops a layer of security, leaving only Elasticsearch's builtin
+<<modules-scripting-other-layers, security layers>>.
+2. Un-sandboxed scripts have unchecked access to Elasticsearch's internals and
+can cause all kinds of trouble if misused.
 
-Less fine-grained settings exist which allow you to enable or disable scripts
-for all sources, all languages, or all contexts.  The following settings
-enable `inline` and `stored` scripts for all languages in all contexts:
 
-[source,yaml]
------------------------------------
-script.inline: true
-script.stored: true
------------------------------------
-
-WARNING:  The above settings mean that anybody who can send requests to your
-Elasticsearch instance can run whatever scripts they choose! This is a
-security risk and may well lead to your Elasticsearch cluster being
-compromised.
+[float]
+[[modules-scripting-other-layers]]
+=== Other security layers
+In addition to user privileges and script sandboxing Elasticsearch uses the
+http://www.oracle.com/technetwork/java/seccodeguide-139067.html[Java Security Manager]
+and native security tools as additional layers of security.
+
+As part of its startup sequence Elasticsearch enables the Java Security Manager
+which limits the actions that can be taken by portions of the code. Painless
+uses this to limit the actions that generated Painless scripts can take,
+preventing them from being able to do things like write files and listen to
+sockets.
+
+Elasticsearch uses
+https://en.wikipedia.org/wiki/Seccomp[seccomp] in Linux,
+https://www.chromium.org/developers/design-documents/sandbox/osx-sandboxing-design[Seatbelt] in mac OS,
+and ActiveProcessLimit on Windows to prevent forking and other undesirable
```

Comment:
I'd found a bunch of links but wasn't familiar enough with the windows API to know which one is canonical. Similarly, I used a link to the chromium docs for Seatbelt because it looked like it explained the mechanism better and actually used the word Seatbelt. I couldn't really follow the Apple links.